kasa/tapo/tapoplug.py

```diff
@@ -0,0 +1,73 @@
+"""Module for a TAPO Plug."""
+import logging
+from datetime import datetime, timedelta
+from typing import Any, Dict, Optional, cast
+
+from ..credentials import Credentials
+from ..emeterstatus import EmeterStatus
+from ..smartdevice import DeviceType
+from .tapodevice import TapoDevice
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TapoPlug(TapoDevice):
+    """Class to represent a TAPO Plug."""
+
+    def __init__(
+        self,
+        host: str,
+        *,
+        port: Optional[int] = None,
+        credentials: Optional[Credentials] = None,
+        timeout: Optional[int] = None,
+    ) -> None:
+        super().__init__(host, port=port, credentials=credentials, timeout=timeout)
+        self._device_type = DeviceType.Plug
+
+    async def update(self, update_children: bool = True):
+        """Call the device endpoint and update the device data."""
+        await super().update(update_children)
+
+        self._energy = await self.protocol.query("get_energy_usage")
+        self._emeter = await self.protocol.query("get_current_power")
+
+        self._data["energy"] = self._energy
+        self._data["emeter"] = self._emeter
+
+        _LOGGER.debug("Got an update: %s %s", self._energy, self._emeter)
```

Comment:
Probably better to log the whole `self._data` here.

Reply:
The base TapoDevice class logs it's self._data so I didn't want to duplicate in the logs.